internal/resources/aws/elasticache_cluster.go

```diff
@@ -104,3 +121,56 @@ func (r *ElastiCacheCluster) backupStorageCostComponent() *schema.CostComponent
 		},
 	}
 }
+
+// ElastiCache implementation of reservationResolver
+type elasticacheReservationResolver struct {
+	term          string
+	paymentOption string
+	cacheNodeType string
+}
+
+func newElasticacheReservationResolver(term, paymentOption, cacheNodeType string) reservationResolver {
+	return &elasticacheReservationResolver{
+		term:          term,
+		paymentOption: paymentOption,
+		cacheNodeType: cacheNodeType,
+	}
+}
+
+func (r elasticacheReservationResolver) IsElasticacheReservedNodeLegacyOffering() bool {
+	for k := range elasticacheReservedNodeCacheLegacyOfferings {
+		if k == r.paymentOption {
+			return true
+		}
+	}
+	return false
+}
+
+func (r elasticacheReservationResolver) PriceFilter() (*schema.PriceFilter, error) {
+	termLength := reservedTermsMapping[r.term]
+	var purchaseOption string
+	if r.IsElasticacheReservedNodeLegacyOffering() {
+		purchaseOption = elasticacheReservedNodeCacheLegacyOfferings[r.paymentOption]
+	} else {
+		purchaseOption = reservedPaymentOptionMapping[r.paymentOption]
+	}
+	validTerms := sliceOfKeysFromMap(reservedTermsMapping)
+	if !stringInSlice(validTerms, r.term) {
+		return nil, fmt.Errorf("Invalid reserved_instance_term, ignoring reserved options. Expected: %s. Got: %s", strings.Join(validTerms, ", "), r.term)
+	}
+	validOptions := append(sliceOfKeysFromMap(reservedPaymentOptionMapping), sliceOfKeysFromMap(elasticacheReservedNodeCacheLegacyOfferings)...)
+
+	if !stringInSlice(validOptions, r.paymentOption) {
+		return nil, fmt.Errorf("Invalid reserved_instance_payment_option, ignoring reserved options. Expected: %s. Got: %s", strings.Join(validOptions, ", "), r.paymentOption)
+	}
+	nodeType := strings.Split(r.cacheNodeType, ".")[1] // Get node type from cache node type. cache.m3.large -> m3
+	if stringInSlice(elasticacheReservedNodeLegacyTypes, nodeType) {
+		log.Warnf("No products found is possible for legacy nodes %s if provided payment option is not supported by the region.", strings.Join(elasticacheReservedNodeLegacyTypes, ", "))
```

Comment:
Is there any way to detect this with the information we have? I'd rather have an explicit validation and return an error.

Reply:
Unfortunately, actually without a dynamic validation, that I guess it is out of scope, this is not possible. 
The problem is that for a specific legacy instance, the available values may differ between regions. 
For example, `cache.m1.small` in  most regions supports `heavy_utilization, medium_utilization, light_utilization`, while in `sa-east-1` only supports `heavy_utilization`. Or `cache.m3.large` only supports `heavy_utilization` in all regions. So the solution I agreed with @vdmgolub was to warn the user about this possible behaviour in the case of reserved legacy instance.
I  described this corner case in the new comment for PriceFilter in elasticache.